internal/servers/repository_workerauth.go

```diff
@@ -0,0 +1,843 @@
+func (r *WorkerAuthRepositoryStorage) Load(ctx context.Context, msg nodee.MessageWithId) error {
+	const op = "workerauth.(WorkerAuthRepositoryStorage).Load"
+	if err := types.ValidateMessage(msg); err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	if msg.GetId() == "" {
+		return errors.New(ctx, errors.InvalidParameter, op, "given message cannot be loaded as it has no ID")
+	}
+
+	marshaledBytes, err := proto.Marshal(msg)
+	if err != nil {
+		return errors.New(ctx, errors.Encode, op, "error marshaling nodee.MessageWithId", errors.WithWrap(err))
+	}
+
+	switch msg.(type) {
+	case *types.NodeInformation:
+		node, err := unmarshalNodeInformation(ctx, marshaledBytes)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		err = r.loadNodeInformation(ctx, node, msg)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+	case *types.RootCertificates:
+		cert, err := unmarshalCertificateAuthority(ctx, marshaledBytes)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		err = r.loadRootCertificates(ctx, cert, msg)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+	default:
+		return errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("message type %T not supported for Load", msg))
+	}
+
+	return nil
+}
+
+// Node information is loaded in two parts:
+// * the workerAuth record
+// * its certificate bundles
+func (r *WorkerAuthRepositoryStorage) loadNodeInformation(ctx context.Context, node *types.NodeInformation, result proto.Message) error {
+	const op = "workerauth.(WorkerAuthRepositoryStorage).loadNodeInformation"
+	if node == nil {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing NodeInformation")
+	}
+
+	authorizedWorker, err := r.findWorkerAuth(ctx, node)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	if authorizedWorker == nil {
+		return nodee.ErrNotFound
+	}
+
+	node.EncryptionPublicKeyBytes = authorizedWorker.WorkerEncryptionPubKey
+	node.CertificatePublicKeyPkix = authorizedWorker.WorkerSigningPubKey
+	node.RegistrationNonce = authorizedWorker.Nonce
+
+	// Default values are used for key types
+	node.EncryptionPublicKeyType = types.KEYTYPE_X25519
+	node.CertificatePublicKeyType = types.KEYTYPE_ED25519
+	node.ServerEncryptionPrivateKeyType = types.KEYTYPE_X25519
+
+	// Decrypt private key
+	databaseWrapper, err := r.kms.GetWrapper(ctx, scope.Global.String(), kms.KeyPurposeDatabase, kms.WithKeyId(authorizedWorker.KeyId))
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	node.ServerEncryptionPrivateKeyBytes, err = decrypt(ctx, authorizedWorker.ControllerEncryptionPrivKey, databaseWrapper)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+
+	// Get cert bundles from the other table
+	certBundles, err := r.findCertBundles(ctx, node.Id)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	node.CertificateBundles = certBundles
+
+	return unmarshalNodeToResult(ctx, node, result)
```

Comment:
Any reason not to just return the node message directly instead of setting it on a proto.Message?